### ISSUE TITLE
Check v6 behavior on the BSDs

### ISSUE THREAD

**Issue by danderson** [tailscale/tailscale#1160]
Now that the core network engine supports IPv6, need to check what the *BSD router implementations do when handed IPv6 addresses and routes, and probably update them to do the right thing based on the OS facilities.

Filing to track this, although I don't currently have the appropriate VMs handy to do this myself.

**Comment by Xe** [comment]
on freebsd and openbsd i am not seeing it creating ipv6 addresses or routes.
openbsd: 522bd88dc217d2bcb5e6d4f000af36eec3b15321ee4d6dc4d2a35c26e8fe4f78
freebsd: 100ff3a8b3ecc636f427fa105361b7406787ddcaf826e09e16f81033c61bee02

I am using git head tailscaled at commit https://github.com/tailscale/tailscale/commit/2d96215d977e9dd823034ff000adbfd1268a4c2b. These are fresh installs of OpenBSD 6.8 and FreeBSD 12.2.

**Comment by Xe** [comment]
TL;DR: no, it doesn't work.

## FreeBSD

FreeBSD 12.2 (log ID 100ff3a8b3ecc636f427fa105361b7406787ddcaf826e09e16f81033c61bee02) with ipv6 address `fd7a:115c:a1e0:ab12:4843:cd96:627b:6075` gets the right network routes:

```
root@ilkash:~ # netstat -r
<snipped>
Internet6:
Destination        Gateway            Flags     Netif Expire
::/96              localhost          UGRS        lo0
localhost          link#2             UH          lo0
::ffff:0.0.0.0/96  localhost          UGRS        lo0
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fd7a:115c:a1e0:ab1 link#3             UH     tailscal
fd7a:115c:a1e0:ab1 tailscale0         UHS    tailscal
fe80::/10          localhost          UGRS        lo0
fe80::%lo0/64      link#2             U           lo0
fe80::1%lo0        link#2             UHS         lo0
ff02::/16          localhost          UGRS        lo0
```

However it fails to set an IP address with `ifconfig`:

```
18 Feb 21 14:38 -0500: router: addr add failed: [ifconfig tailscale0 inet6 fd7a:115c:a1e0:ab12:4843:cd96:627b:6075/128 fd7a:115c:a1e0:ab12:4843:cd96:627b:6075] => exit status 1
ifconfig: ioctl (SIOCAIFADDR): File exists
```

Additionally, I am unable to produce a variant of this command that works (including trying to play with prefixlen and alias flags). I keep getting the `File exists` error.

## OpenBSD

OpenBSD 6.8 (log id `522bd88dc217d2bcb5e6d4f000af36eec3b15321ee4d6dc4d2a35c26e8fe4f78`) fails to get an ipv4 address set entirely, with no logs about this failure either:

![Screenshot_openbsd6 7_2021-02-18_15:02:22](https://user-images.githubusercontent.com/529003/108415136-1c45d280-71fb-11eb-86f1-3027ed7690ce.png)

However when you try to ping it, `tailscaled` does actually report that it's trying to process an ICMP message:

```18 Feb 21 15:08 -0500: Accept: ICMPv4{100.67.184.57:0 > 100.123.95.22:0} 84 icmp ok
```

**Comment by DentonGentry** [comment]
I think this is done:

- FreeBSD fixed in #1307 
- OpenBSD fixed in #1372
- HardenedBSD identifies itself as FreeBSD, #1307 works for it too
- NetBSD is out of scope for this effort.
